CSharp/BatchExplorer/ViewModels/MainViewModel.cs

```diff
@@ -733,6 +733,95 @@ public CommandBase RefreshJobs
         }
 
         /// <summary>
+        /// Delete selected jobs
+        /// </summary>
+        public CommandBase DeleteSelectedJobs
+        {
+            get
+            {
+                return new CommandBase( 
+                    async (commandArg) =>
+                    {                        
+                        if (this.jobs.Any(a => a.IsChecked))
+                        {
+                            IEnumerable<string> deletedIds = this.jobs.Where(a => a.IsChecked).Select(s=>s.Id);
+                            var result =
+                            MessageBox.Show(
+                                deletedIds.Count() > 1 ? 
+                                $"Are you sure you want to delete {deletedIds.Count()} selected jobs?" :
+                                "Are you sure you want to delete the job?",
+                                "Confirm Delete",
+                                MessageBoxButton.YesNo);
+
+                            if (result == MessageBoxResult.Yes)
+                            {
+                                this.LeftSpinnerIsVisible = true;
+                                foreach (var job in this.jobs)
+                                {
+                                    if (job.IsChecked)
+                                    {
+                                        await job.DeleteAsync(false);
+                                    }
+                                }
+
+                                //This is very hacky as the tab count never updated correctly if we refresh
+                                //just after issueing delete async inspite of awaiting for result. Probably it is by default nature
+                                //So keep on refreshing until we see the deleted record inside the refresh list
+                                bool requiredRefresh = true;

```

Comment:
Instead of removing this logic, I have implemented Timeout feature. The loop will execute at max 1 min after that UI become unblocked even there is no update of the tab count. Please check the   latest commit.